### PR TITLE
[SR-4078] Provide Sort Operator

### PR DESCRIPTION
**Pull request by Pslydhh**
Provide Sort Operator for pipeline:

- Keep TopNNode for original volcano model executor.
- Add pull_chunk/finish at ChunksSorter for sort operator.
- Use update of ChunksSorter to collect chunk and sort.
- No statistics have been added at present.

### CODE REVIEW

**Review by liuyehcf — DISMISSED** [review]

**Review by satanson — APPROVED** [review]

File: be/src/exec/pipeline/sort/sort_sink_operator.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    // This method is the same as topn node.
+    vectorized::ChunkPtr _materialize_chunk_before_sort(vectorized::Chunk* chunk);
+    bool _is_finished = false;
+
+    std::shared_ptr<vectorized::ChunksSorter> _chunks_sorter;
+
+    // from topn
+    // _sort_exec_exprs contains the ordering expressions
+    SortExecExprs _sort_exec_exprs;
+    std::vector<OrderByType> _order_by_types;
```
Comment (kangkaisen) [inline be/src/exec/pipeline/sort/sort_sink_operator.h:67]: Could use const reference

File: be/src/exec/pipeline/sort/sort_source_operator.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+namespace pipeline {
+class SortSourceOperator final : public SourceOperator {
+public:
+    SortSourceOperator(int32_t id, int32_t plan_node_id, std::shared_ptr<vectorized::ChunksSorter>&& chunks_sorter)
+            : SourceOperator(id, "sort_source", plan_node_id), _chunks_sorter(chunks_sorter) {}
+
+    ~SortSourceOperator() override = default;
+
+    // Result sink will send result chunk to BufferControlBlock directly,
```
Comment (kangkaisen) [inline be/src/exec/pipeline/sort/sort_source_operator.h:29]: Comment is wrong

File: be/src/exec/pipeline/sort/sort_source_operator.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    // Then FE will pull result from BufferControlBlock
+    bool has_output() override;
```
Comment (kangkaisen) [inline be/src/exec/pipeline/sort/sort_source_operator.h:31]: Comment is wrong

File: be/src/exec/vectorized/topn_node.cpp [hunk 4]
```diff
@@ -211,4 +214,42 @@ ChunkPtr TopNNode::_materialize_chunk_before_sort(Chunk* chunk) {
...
+        chunks_sorter = std::make_unique<vectorized::ChunksSorterTopn>(&(_sort_exec_exprs.lhs_ordering_expr_ctxs()),
+                                                                       &_is_asc_order, &_is_null_first, _offset, _limit,
+                                                                       SIZE_OF_CHUNK_FOR_TOPN);
+    } else {
+        chunks_sorter = std::make_unique<vectorized::ChunksSorterFullSort>(&(_sort_exec_exprs.lhs_ordering_expr_ctxs()),
+                                                                           &_is_asc_order, &_is_null_first,
+                                                                           SIZE_OF_CHUNK_FOR_FULL_SORT);
+    }
+
+    //chunks_sorter->setup_runtime(mem_tracker(), runtime_profile(), "ChunksSorter");
```
Comment (kangkaisen) [inline be/src/exec/vectorized/topn_node.cpp:237]: Remove unused code.

File: be/src/exec/pipeline/sort/sort_sink_operator.h (hunk 1 above)
Comment (Pslydhh) [reply be/src/exec/pipeline/sort/sort_sink_operator.h:67]: It couldn't use const reference

**Pull request closed without merging**